src/ape/api/projects.py

```diff
@@ -153,7 +153,8 @@ def contracts(self) -> Dict[str, ContractType]:
 
     @property
     def _cache_folder(self) -> Path:
-        folder = self.contracts_folder.parent / ".build"
+        current_ecosystem = self.network_manager.network.ecosystem.name
+        folder = self.contracts_folder.parent / ".build" / current_ecosystem
```

Comment:
what is needed for ethpm-types ?

Reply:
Currently the data structure is `ContractType.name -> `ContractType`, but doesn't have a way to delineate between different productions of bytecode using different VM rules/compiler config

Should be something more like `ContractType.*bytecode` fields should allow multiple bytecode productions, indexed by compiler settings e.g. `ContractType.runtimeBytecode[Compiler.name] -> Bytecode`
https://github.com/ApeWorX/ethpm-types/blob/main/ethpm_types/contract_type.py#L79

This may take some thinking

Reply:
The other option is back to what I said before, there's a `.build/{ecosystem}/` folder that the ecosystem plugin creates mirror copies of the types that actually end up in the published/cached contract manifest, and those mirrors are kept in sync every time you try to do something on that ecosystem. since, at least in this instance, the goal is that you are using the same source code for that ecosystem, it's not really necessary to have the transpiled code tracked alongside the EVM code (or add it as an additional field to `ethpm_types.ContractType`, which should accept additional properties not defined in the model)

Reply:
I couldnt really understand this. Manifest and abi files will be synced as we have network option and usage now. Can you give more details on this? What should i do?